Cython/Compiler/Nodes.py

```diff
@@ -1718,8 +1718,14 @@ def analyse_declarations(self, env):
         if self.items is not None:
             if self.in_pxd and not env.in_cinclude:
                 self.entry.defined_in_pxd = 1
+            # For extern enums we can't reason about their equivalent int values because
+            # we can't see them.
+            last_enum_value = 0 if self.visibility != 'extern' else None
             for item in self.items:
-                item.analyse_declarations(scope, self.entry)
+                value_entry = item.analyse_declarations(scope, self.entry, last_enum_value)
+                last_enum_value = value_entry.equivalent_enum_value if self.visibility != 'extern' else None
```

Comment:
Can't `equivalent_enum_value` simply be `None` if `visibility == 'extern'`? That would avoid the additional condition.

Reply:
It looks like you can specify a value for elements in extern enums:

```
cdef extern from *:
    enum E:
        A = 1
        B
```

In this case I think it's reasonable for Cython to trust what the user says about `A` and use it as `equivalent_enum_value`, but not to infer anything about `B` from this.

Your proposal would work though - it's not using what we've been told about `A`